modules/affile/affile-dest.c

```diff
@@ -191,8 +203,12 @@ affile_dw_reopen(AFFileDestWriter *self)
                 evt_tag_str("filename", self->filename),
                 evt_tag_errno(EVT_TAG_OSERROR, errno),
                 NULL);
+      if(!self->reopen_pending)
+        affile_dw_arm_reopen(self);
+      g_static_mutex_unlock(&self->lock);
       return self->owner->super.super.optional;
     }
+  g_static_mutex_unlock(&self->lock);
   return TRUE;
 }
 
```

Comment:
* That's true. It was dumb to left those locks/releases there. 
* I will run checks on that.
* That's also true but I think we should separate time-reopen and time-reap functionalities. The first one is responsible for reopening a dead connections, the second for close the connection if it's idle. They can have different time parameters, they are used in different states of a affile_dw. Of course we can use one timer for reap and reopen to half the number of them and reregister the handler function in different states. 

I'll rethink my solution and try to eliminate those locks.